jina/peapods/runtimes/head/__init__.py

```diff
@@ -209,6 +211,12 @@ async def _handle_data_request(
                 requests, pod='uses_before'
             )
             requests = [response]
+        elif len(requests) > 1:
+            docs_matrix = get_docs_matrix_from_request(requests, field='docs')
```

Comment:
I would suggest to move this and the following line to the DataRequestHandler as a static function? Then there is less doc related logic here in this runtime and we dont have to copy those two lines again in line 244/245

Reply:
I refactored this, can you re-check ?